src/main/java/com/cloudbees/jenkins/plugins/bitbucket/filesystem/BitbucketSCMFile.java

```diff
@@ -24,40 +24,61 @@
 
 package com.cloudbees.jenkins.plugins.bitbucket.filesystem;
 
+import com.cloudbees.jenkins.plugins.bitbucket.BranchSCMHead;
+import com.cloudbees.jenkins.plugins.bitbucket.JsonParser;
+import com.cloudbees.jenkins.plugins.bitbucket.PullRequestSCMHead;
+import java.util.Map;
+
 import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketApi;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.IOException;
 import java.io.InputStream;
 import jenkins.scm.api.SCMFile;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.mixin.ChangeRequestCheckoutStrategy;
 
 public class BitbucketSCMFile  extends SCMFile {
 
     private final BitbucketApi api;
-    private  String ref;
+    private SCMHead head;
     private final String hash;
 
     public String getRef() {
-        return ref;
-    }
+        if (head instanceof BranchSCMHead) {
+            return head.getName();
+        }
+        if (head instanceof PullRequestSCMHead) {
```

Comment:
This still won't work for bitbucket cloud: https://bitbucket.org/site/master/issues/5814/refify-pull-requests-by-making-them-a-ref

Reply:
Not sure on how the API usage should look like for merges on bitbucket cloud. I've setup a repo on bitbucket to explore this further: https://bitbucket.org/seb-lie/test/src/master/